### DISCUSSION TITLE
Rpi zero w  as mqtt transmitter

### DISCUSSION THREAD

**Discussion by Jackzerox** [syssi/esphome-jk-bms#407] [Q&A]
Hi, thanks for your project, Im trying to figure out how to use a rpi zero w to communicate via bluetooth and get the data from the jk bms and transmit the data via mqtt, in this rpi I have nodered installed, and communicate with other rpi in other place via zerotier vpn.

It's this possible ?

**Comment by syssi** [comment]
These are the supported platforms/microcontrollers by ESPHome: https://esphome.io/#platforms

It looks like you are looking for a Linux application to talk to the BMS. Did you try the `mppsolar` project already?

**Comment by Jackzerox** [comment]
Thanks for you reply, I just found a similar project using python and running on raspberry pi , but I'm still want the possibility to use the esp32 (https://github.com/fl4p/batmon-ha/tree/master), other question, can I use websocket mqtt with esp32 on your project ?

**Reply by syssi** [reply]
The connection between the ESP and the MQTT broker will be an ordinary MQTT connection. If your MQTT broker supports MQTT via websockets you will be able to talk to your broker using your webbroser. Is this what you are looking for?
